doc/source/contributing/docstrings.md

```diff
@@ -0,0 +1,68 @@
+(docstrings)=
+# Docstring formatting and type hints
+
+Docstrings should follow the
+[numpy docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html).
+Extra guidance can also be found in
+[pandas docstring guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html).
+Please reasonably document any additions or changes to the codebase,
+when in doubt, add a docstring.
+
+The different formatting and aim between numpydoc style type description and
+[type hints](https://docs.python.org/3/library/typing.html)
+should be noted. numpydoc style targets docstrings and aims to be human
+readable whereas type hints target function definitions and `.pyi` files and
+aim to help third party tools such as type checkers or IDEs. ArviZ does not
+require functions to include type hints
+however contributions including them are welcome.
+
+## References
+Docstrings follow the same guide for adding references as the other docs.
+For adding refernces to external libraries functions and objects, see {ref}`reference_external_libs`. For referencing ArviZ objects, follow {ref}`reference_arviz_objects`.
+
+## See Also
+For adding the _See Also_ docstring section, you just need to add the function name. Sphinx will
+automatically add links to other ArviZ objects and functions listed in the _See Also_
+section.
+
+For example, let's add {func}`~arviz.hdi` and {func}`~arviz.plot_ppc` in the _See Also_ section.
+
+```
+    See Also
+    --------
+    hdi : Calculate highest density interval (HDI) of array for given probability.
+    plot_ppc : plot for posterior/prior predictive checks.
+```
+
+## Kwargs parameters
+All the kwargs parameters in {ref}`plots <plot_api>` modules are passed to the matplotlib or bokeh functions. While writing their description, the functions to which they are being passed must be mentioned. The process for checking those functions is similar for all the kwargs arguments. Let's read the step-by-step guide for `backend_kwargs`.
+
+### Backend Kwargs
```

Comment:
might be a bit of a side comment, but I think we should discuss that anyway. I envisioned this as a guide for developers/maintainers/arviz teammembers/<other alias>, with the more basic content on the above contributor guide. The steps below however are aimed at people with little to no python knowledge, I think that for the intended audience of this doc, 1-2 sentences explaining that the functions where kwargs are eventually passed to must be referenced should be enough.